### PR TITLE
Migrate Ember provider to new structure

### PR DESCRIPTION
**Pull request by rossf7**
Towards #44

### CODE REVIEW

File: pkg/internal/data/ember.go [hunk 1]
```diff
@@ -1,60 +1,17 @@
...
-	if !ok {
-		return nil, fmt.Errorf("country code %q not found", countryCode)
-	}
-
-	return &result, nil
-}
-
-func getGridIntensityData() (map[string]GridIntensity, error) {
-	data := map[string]GridIntensity{}
+func GetEmberGridIntensity() (map[string]EmberGridIntensity, error) {
```
Comment (rossf7) [inline pkg/internal/data/ember.go:13]: This new `pkg/internal/data` package means the data and logic for getting the data file is separate from the provider.

We could extend this for other data files or to add more Ember data. The internal name is so its clear this isn't part of the public API.

File: pkg/provider/ember.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+func (a *EmberClient) GetCarbonIntensity(ctx context.Context, region string) ([]CarbonIntensity, error) {
+	region = strings.ToUpper(region)
+	result, ok := a.data[region]
+	if !ok {
+		return nil, fmt.Errorf("region %q not found", region)
+	}
+
+	validFrom := time.Date(emberDataYear, 1, 1, 0, 0, 0, 0, time.UTC)
+	validTo := time.Date(emberDataYear, 12, 31, 23, 59, 0, 0, time.UTC)
```
Comment (rossf7) [inline pkg/provider/ember.go:41]: Here we set the time period for the data.

File: pkg/internal/data/ember.go (hunk 1 above)
Comment (mrchrisadams) [reply pkg/internal/data/ember.go:13]: I hadn't thought of this as an idea, but I really like it. It's also something we could use on any other libraries that use local data. 👍

**Comment by mrchrisadams** [comment]
hey @rossf7  - this is good to merge in. Thanks!

**Review by mrchrisadams — APPROVED** [review]

**Pull request merged**